### PR TITLE
[FIX] product,purchase,sale_*: create default product catalog kanban record

### PR DESCRIPTION
**Pull request by bastvdn**
The sale_stock module patches the ProductCatalogKanbanRecord updateQuantity method to prevent the quantity to go under the delivered quantity of the product

The issue is that we are also using the ProductCatalogKanbanRecord in the industry_fsm_stock module we are also doing an override of the updateQuantity method to prevent the quantity of a product to go under the 'minimumQuantityOnProduct' which can be lower that the delivered quantity of a product in the case of a service product

Task-3689939

https://github.com/odoo/enterprise/pull/55455

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by damrOdoo — COMMENTED** [review]
looks good to me. Thanks for your fix. Just the same comment as in the enterprise part. If this is an imp, target master. Otherwise, update your commit title.

**Comment by bastvdn** [comment]
@damrOdoo 
Thanks for the review,
sorry for the delay, check my changes

**Review by auon-odoo — APPROVED** [review]
Lgtm :+1:

File: addons/sale_stock/static/src/product_catalog/kanban_record.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
 /** @odoo-module */
-import { ProductCatalogKanbanRecord } from "@product/product_catalog/kanban_record";
+import { ProductCatalogKanbanRecord, DefaultProductCatalogKanbanRecord } from "@product/product_catalog/kanban_record";
 import { ProductCatalogSaleOrderLine } from "./sale_order_line/sale_order_line";
 import { patch } from "@web/core/utils/patch";
 
+patch(DefaultProductCatalogKanbanRecord.prototype, {
```
Comment (xavierbol) [inline addons/sale_stock/static/src/product_catalog/kanban_record.js:6]: And if we pass in the `productCatalogData` the type of the product to know if it is a storable product, then we could just apply the `updateQuantity` patch in that file only for those products type, no? 🤔

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I just left one question before asking a review to Sales team. 🙂

File: addons/sale_stock/static/src/product_catalog/kanban_record.js (hunk 1 above)
Comment (bastvdn) [reply addons/sale_stock/static/src/product_catalog/kanban_record.js]: @xavierbol mmh yeah this should be simpler, check my fix and tell me what you think

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂

File: addons/sale/models/sale_order.py [hunk 2]
```diff
@@ -1823,6 +1823,7 @@ def _get_product_catalog_order_data(self, products, **kwargs):
         res = {}
         for product in products:
             res[product.id] = {'price': pricelist.get(product.id)}
+            res[product.id]['productType'] = product.type
```
[the remark this answers is not in the document]
Comment (bastvdn) [reply addons/sale/models/sale_order.py:1826]: kk but i'll still need to add the productType for product who aren't in the record yet so i still add it in the sale_order _get_product_catalog_order_data() override
Tell me what you think

[the remark this answers is not in the document]
Comment (chevalierv) [reply addons/sale/models/sale_order.py:1828]: Indeed, but it wouldn't solve the issue of other modules having to define it. Could you try to modify `_get_product_catalog_order_data` directly in the mixin to return the product type?

[the remark this answers is not in the document]
Comment (bastvdn) [reply addons/sale/models/sale_order.py]: @chevalierv done, check it out

**Review by chevalierv — APPROVED** [review]
Hello @bastvdn 

LGTM :+1: Thanks for improving the catalog!

**Review by xavierbol — COMMENTED** [review]
robodoo r+

**Pull request closed without merging**